pinot-controller/src/main/resources/app/components/User/AddUser.tsx

```diff
@@ -0,0 +1,242 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  Checkbox,
+  DialogContent,
+  FormControl,
+  Input,
+  InputLabel, ListItemText,
+  MenuItem,
+  Select,
+  TextField,
+  InputAdornment,
+  IconButton,
+} from '@material-ui/core';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+type Props = {
+  tableList: Array<{
+    name: string,
+    checked: boolean,
+    disabled: boolean
+  }>,
+  setUserInfo: Function,
+  userProp?: {
+    username: string,
+    password: string,
+    component: string,
+    role: string,
+    tables: Array<string>,
+    permissions: Array<string>
+  }
+}
+
+export default function AddUser({
+  tableList,
+  setUserInfo,
+  userProp
+}: Props) {
+  const PERMISSIONS = ["READ", "CREATE", "UPDATE", "DELETE"];
+  const [user, setUser] = useState({
+      ...userProp,
+      tables: ["DUAL"]
+  });
+  const [tables, setTables] = useState(tableList);
+  const [showPassword, setShowPassword] = useState(false);
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+  const handleMouseDownPassword = (event) => {
+    event.preventDefault();
+  };
+  const changeHandler = (field, e)=>{
+    let val = e.target.value;
+    let userCopy = {...user};
```

Comment:
Spread operator usually creates a shallow copy which creates problems down the road. can we please use lodash deepClone?

Reply:
You are right. Thanks for your suggestion.  I will fix it later.